toolkits/landcover/test/unit/named-args.unit.test.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var NamedArgs = require('../../impl/named-args').NamedArgs;
+
+// Test function acts as a thin wrapper, returning exactly what is returned
+// by extractFromFunction for testing.
+var testFunction = function(foo, bar, baz) {
+  return NamedArgs.extractFromFunction(testFunction, arguments);
+};
+
+withEarthEngineStub('NamedArgs', function() {
+  it('extractFromFunction() builds dictionary', function() {
+    var actual = testFunction(3, 141, 59);
+    var expected = {foo: 3, bar: 141, baz: 59};
+    expect(actual).toEqual(expected);
+  });
+
+  it('extractFromFunction() returns dictionary if specified', function() {
+    var actual = testFunction({foo: 3, bar: 141, baz: 59});
+    var expected = {foo: 3, bar: 141, baz: 59};
+    expect(actual).toEqual(expected);
+  });
+
+  it('extractFromFunction() throws error on unsupported fn declaration',
```

Comment:
You should explicitly test fewer/too many/out-of-order args as well as passing null and undefined.

Some additional signatures I think you should test:
function a() { return 1; }            // no args
function a(c, d) { return b(e, f); }          // 1-line
function a(c, /* comment */ d) { return b(e, f); }   // block comment in args
function a(\nb, // inline comment\nc) { return b(e,f); }   // 1-line comment in args

Reply:
Don't want to test these?

Reply:
Done. PTAL.

Also, what do you mean "out of order" args here?

Reply:
I was thinking you happen to be testing a function with args (foo, bar, baz) with a dictionary with args in the same order, but they don't really have an order, do they?  So it doesn't matter.

Reply:
AAh ok, yes, the order of dictionaries in JavaScript is prescribed by the impl, not by the user, so as far as the user is concerned, in principle there's no fixed order.